src/main/java/org/molgenis/vcf/decisiontree/filter/VcfRecord.java

```diff
@@ -67,47 +72,185 @@ public Object getValue(Field field, Allele allele, Integer sampleIndex) {
       case INFO_VEP:
         value = getNestedVepValue(field);
         break;
+      case FORMAT_GENOTYPE:
+        value = sampleContext != null ? getNestedGTValue((NestedField) field, sampleContext) : null;
+        break;
       case FORMAT:
-        if (sampleIndex == null) {
+        if (sampleContext == null) {
           throw new UnsupportedOperationException(
               "Cannot filter on FORMAT fields when running in variant filter mode.");
         }
-        value = getFormatField(field, sampleIndex);
+        value = getFormatField(field, sampleContext);
+        break;
+      case SAMPLE:
+        value = sampleContext != null ? getSampleValue(field, sampleContext) : null;
         break;
       default:
         throw new UnexpectedEnumException(fieldType);
     }
     return value;
   }
 
-  private Object getFormatField(Field field, Integer sampleIndex) {
-    Genotype genotype = variantContext.getGenotype(sampleIndex);
+  private Object getSampleValue(Field field, SampleContext sampleContext) {
+    Object value;
+    SampleFieldType sampleField = SampleFieldType.valueOf(field.getId().toUpperCase());
+    switch (sampleField) {
+      case AFFECTED_STATUS:
+        value = sampleContext.getAffectedStatus().toString();
+        break;
+      case PROBAND:
```

Comment:
I think it is clearest to treat all PED file fields the same way. I can imagine use cases where you might want to have a tree to classify all samples in case of affected parents for instance, but treat the probands and non-probands in a different way.

Reply:
Correction, proband is not a part of the PED off course... so removing it sounds fair